WooCommerce/Classes/ServiceLocator/ProductImageUploader.swift

```diff
@@ -302,15 +403,31 @@ private extension ProductImageUploader {
                                                             productOrVariationID: key.productOrVariationID,
                                                             error: .failedUploadingImage(asset: asset, error: error))
                 if statusUpdatesExcludedProductKeys.contains(key) == false {
-                    errorsSubject.send(infoError)
+                    if featureFlagService.isFeatureFlagEnabled(.backgroundProductImageUpload) {
+                        let failedStatus = ProductImageStatus.uploadFailure(asset: asset,
+                                                                            error: error,
+                                                                            siteID: key.siteID,
+                                                                            productID: key.productOrVariationID)
+                        self.userDefaultsStatuses.updateStatus(failedStatus)
```

Comment:
Updating the status in the storage is unnecessary here since you were already receiving status updates in `observeStatusUpdates`.

Reply:
Good catch Huong! This created a duplicated storage update for the same failure event. I tested it and it didn't create race condition or anything, but we definitely update the state unnecessarily (adding more writes to the file system).

So, I updated it here ff2a952b6efcfd26ca952d2573f90e39064b1cba adding also some reminders for the future.